src/wp-includes/class-wp-block.php

```diff
@@ -333,7 +333,44 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		switch ( $block_type->attributes[ $attribute_name ]['source'] ) {
 			case 'html':
 			case 'rich-text':
-				$block_reader = new WP_HTML_Tag_Processor( $block_content );
+				// Create private anonymous class until the HTML API provides `set_inner_html` method.
```

Comment:
@gziolo @dmsnell I extended the `WP_HTML_Tag_Processor` class this way to include a private `set_inner_text` method until this is possible with the core API. I made this to avoid using a regex. I couldn't find a similar approach in the code. Is this acceptable or should I look for an alternative approach?